src/__tests__/engine/classic/sign_up_pane.test.jsx

```diff
@@ -44,6 +56,50 @@ describe('SignUpPane', () => {
 
     expectComponent(<Component {...defaultProps} instructions="instructions" />).toMatchSnapshot();
   });
+
+  it('shows the Captcha pane', () => {
+    require('core/index').captcha.mockReturnValue({
+      get() {
+        return true;
+      }
+    });
+
+    require('engine/classic').isSSOEnabled.mockReturnValue(false);
+
+    const Component = getComponent();
+
+    expectShallowComponent(<Component {...defaultProps} />).toMatchSnapshot();
+  });
+
+  it('hides the Captcha pane for SSO connections', () => {
+    require('core/index').captcha.mockReturnValue({
+      get() {
+        return true;
+      }
+    });
+
+    require('engine/classic').isSSOEnabled.mockReturnValue(true);
+
+    const Component = getComponent();
+
+    expectShallowComponent(<Component {...defaultProps} />).toMatchSnapshot();
+  });
+
+  it('shows the Captcha pane for SSO (ADFS) connections', () => {
```

Comment:
Another one that goes thru the same endpoint and is enterprise is the "ad" strategy... Are we also handling that case?

Reply:
The logic here relies on the result of `isHRDDomain`, which looks like it checks both the `ad` and `auth0-adldap` strategies:

https://github.com/auth0/lock/blob/c692eb638543d4b85a740b19fd639ec861ef37b4/src/connection/enterprise.js#L142

Is this sufficient?

Reply:
mak sense!